rust/chains/hyperlane-sealevel/src/mailbox.rs

```diff
@@ -0,0 +1,754 @@
+        &self,
+        instruction: Instruction,
+    ) -> ChainResult<Option<T>> {
+        simulate_instruction(
+            &self.rpc_client,
+            self.payer
+                .as_ref()
+                .ok_or_else(|| ChainCommunicationError::SignerUnavailable)?,
+            instruction,
+        )
+        .await
+    }
+
+    /// Simulates an Instruction that will return a list of AccountMetas.
+    pub async fn get_account_metas(
+        &self,
+        instruction: Instruction,
+    ) -> ChainResult<Vec<AccountMeta>> {
+        get_account_metas(
+            &self.rpc_client,
+            self.payer
+                .as_ref()
+                .ok_or_else(|| ChainCommunicationError::SignerUnavailable)?,
+            instruction,
+        )
+        .await
+    }
+
+    /// Gets the recipient ISM given a recipient program id and the ISM getter account metas.
+    pub async fn get_recipient_ism(
+        &self,
+        recipient_program_id: Pubkey,
+        ism_getter_account_metas: Vec<AccountMeta>,
+    ) -> ChainResult<Pubkey> {
+        let mut accounts = vec![
+            // Inbox PDA
+            AccountMeta::new_readonly(self.inbox.0, false),
+            // The recipient program.
+            AccountMeta::new_readonly(recipient_program_id, false),
+        ];
+        accounts.extend(ism_getter_account_metas);
+
+        let instruction = Instruction::new_with_borsh(
+            self.program_id,
+            &hyperlane_sealevel_mailbox::instruction::Instruction::InboxGetRecipientIsm(
+                recipient_program_id,
+            ),
+            accounts,
+        );
+        let ism = self
+            .simulate_instruction::<SimulationReturnData<Pubkey>>(instruction)
+            .await?
+            .ok_or(ChainCommunicationError::from_other_str(
+                "No return data from InboxGetRecipientIsm instruction",
+            ))?
+            .return_data;
+        Ok(ism)
+    }
+
+    /// Gets the account metas required for the recipient's
+    /// `MessageRecipientInstruction::InterchainSecurityModule` instruction.
+    pub async fn get_ism_getter_account_metas(
+        &self,
+        recipient_program_id: Pubkey,
+    ) -> ChainResult<Vec<AccountMeta>> {
+        let (account_metas_pda_key, _) = Pubkey::find_program_address(
+            hyperlane_sealevel_message_recipient_interface::INTERCHAIN_SECURITY_MODULE_ACCOUNT_METAS_PDA_SEEDS,
+            &recipient_program_id,
+        );
+        let instruction =
+            hyperlane_sealevel_message_recipient_interface::MessageRecipientInstruction::InterchainSecurityModuleAccountMetas;
+        let instruction = Instruction::new_with_bytes(
+            recipient_program_id,
+            &instruction
+                .encode()
+                .map_err(ChainCommunicationError::from_other)?,
+            vec![AccountMeta::new(account_metas_pda_key, false)],
+        );
+
+        self.get_account_metas(instruction).await
```

Comment:
I'm seeing a repeated pattern in a few of the methods here: find the program address, create `Instruction::new_with_bytes`, then return `self.get_account_metas(instruction).await`. This could be refactored into a method that take the `account_metas_pda_key`, `instruction`, and any other missing arguments as parameters.